crates/sui-node/src/lib.rs

```diff
@@ -384,58 +387,26 @@ impl SuiNode {
             epoch_store.epoch_start_state(),
         )
         .expect("Initial trusted peers must be set");
-        let state_archive_handle = if let Some(remote_store_config) =
-            &config.state_archive_write_config.object_store_config
-        {
-            let local_store_config = ObjectStoreConfig {
-                object_store: Some(ObjectStoreType::File),
-                directory: Some(config.archive_path()),
-                ..Default::default()
-            };
-            let archive_writer = ArchiveWriter::new(
-                local_store_config,
-                remote_store_config.clone(),
-                FileCompression::Zstd,
-                StorageFormat::Blob,
-                Duration::from_secs(600),
-                256 * 1024 * 1024,
-                &prometheus_registry,
-            )
-            .await?;
-            Some(archive_writer.start(state_sync_store).await?)
-        } else {
-            None
-        };
-        let db_checkpoint_config = if config.db_checkpoint_config.checkpoint_path.is_none() {
-            DBCheckpointConfig {
-                checkpoint_path: Some(config.db_checkpoint_path()),
-                ..config.db_checkpoint_config.clone()
-            }
-        } else {
-            config.db_checkpoint_config.clone()
-        };
 
-        let db_checkpoint_handle = match db_checkpoint_config
-            .checkpoint_path
-            .as_ref()
-            .zip(db_checkpoint_config.object_store_config.as_ref())
-        {
-            Some((path, object_store_config)) => {
-                let handler = DBCheckpointHandler::new(
-                    path,
-                    object_store_config,
-                    60,
-                    db_checkpoint_config
-                        .prune_and_compact_before_upload
-                        .unwrap_or(true),
-                    config.indirect_objects_threshold,
-                    config.authority_store_pruning_config,
-                    &prometheus_registry,
-                )?;
-                Some(handler.start())
-            }
-            None => None,
-        };
+        // Start archiving local state to remote store
+        let state_archive_handle =
+            Self::start_state_archival(&config, &prometheus_registry, state_sync_store).await?;
+
+        // Start uploading state snapshot to remote store
+        let state_snapshot_handle = Self::start_state_snapshot(
+            &config,
+            &prometheus_registry,
+            !epoch_store
+                .protocol_config()
+                .simplified_unwrap_then_delete(),
```

Comment:
Not strongly opinionated here, but it feels like a waste to me to pipe this one protocol config flag from here all the way through to the snapshot writer. It makes more sense to me to pass the protocol config itself, that way if there are any other protocol config flags that are relevant to snapshot writer or uploader in the future, we don't need to re-pipe everything

Reply:
Makes sense, it'd definitely make a lot more sense to pipe it through once we have another usage of protocol config.

Reply:
maybe a n00b question: variable is set once during the startup and never changes, right? So if you start backup process on a fresh node it'll stay `false` forever and even latest state snapshots will be uploaded without simplified unwrap_then_delete. Or am I missing something?

Reply:
Actually that's not right, we need to fetch the protocol config from latest epoch store. Let me make that change